### ISSUE TITLE
All follow requests on some instances are stuck as pending, including unlocked accounts

### ISSUE THREAD

**Issue by MirceaKitsune** [tootsuite/mastodon#12431]
When following any user on an affected instance, you're told that your follow request is pending. However this happens for all accounts, including ones that aren't locked and should accept follow requests immediately! I can favorite boost or comment on their posts just fine, but I cannot follow them. Both the local and target instance(s) are running Mastodon v3.0.1.

I tested this with a bot I own on an instance that has this issue: On the bot account I see no pending request nor any change, but on my main account I always see the "cancel follow request" button.

**Comment by Izzz31** [comment]
Wanted to report the exact same issue.
I'm the admin of mastodon.izzz.fr (runnning v3.0.1) and I'm experiencing this with at least mastodon.social, mastodon.cloud and mastodon.gougere.fr

In my case it comes with an other issue : I don't have posts from followed accounts in my instance's feeds for about 10 days. I suspect some kind of outdated push subscriptions but don't know how to refresh them.

**Comment by MirceaKitsune** [comment]
Since I have alt accounts on several instances, I ran a few more tests on this problem. I went to a second instance I'm registered on, then proceeded to follow an account from one of the affected instances: I got the exact same issue, the request was marked as pending and remains that way forever. Next I went to a third instance where I have an account, and followed the same account from the affected instance: From here I had no issue and could follow it immediately. All three instances involved in this test are running v3.0.1.

I tried following affected accounts both locally (searching the account from my local instance then clicking its follow icon) and remotely (going to the account's page on its instance then letting the popup window and ask from which instance I want to preform the action). The issue occurs in both cases, even when the remote follow shows the message "Success! You are now following: Name"

**Comment by trwnh** [comment]
@Izzz31 In your case, I tried to lookup your account from mastodon.social and got an SSL certificate error.

@MirceaKitsune I can't say for sure but you might want to consider the possibility that some of your accounts may be silenced on certain instances

**Comment by MirceaKitsune** [comment]
@trwnh Not the case: I spoke with the admin of one instance that has this problem, he's also confused about what's happening and we've been investigating it. I could also reproduce it between two accounts I own on different instances.

**Comment by c-chell** [comment]
I try to reproduce the problem, but with a test to and from my instance with my account, I see no problem, from a remote instance to my instance (, no problem.
So I conclude the problem can affect only some users on instance.
Also, even I block some instance, no instance, from where the user have the problem, is blocked and I have no user block on my account nor my instance.
I don't know what to see and to check the problem.

Can it be due to a limit of requests allow per time ? (I haven't set any of this type of limit other than my server is a little "too busy" and have some lag)

**Comment by Izzz31** [comment]
@trwnh Ok problems solved for me. That was due to SSL configuration, I switched back to TLS 1.2 from TLS 1.3
Just found the same issue here #10866
Thanks for help!

**Comment by c-chell** [comment]
Little update, I just receive a notificaiton to accept a follow even if my account is not configured to do that. The only reason I see is the instance is masked (to avoid public timeline content).

**Comment by MirceaKitsune** [comment]
@Izzz31 If it's a SSL issue, doesn't that mean I shouldn't be able to interact with the instance and affected accounts at all? I can favorite boost or reply fine, only following is broken. Perhaps sending and / or receiving follow requests works in a different so only it triggers this issue?

**Comment by Izzz31** [comment]
@MirceaKitsune Is it working both ways? In my case I was able to fav/boost/reply from my instance but any interaction from affected instances to my instance was not federated.

**Comment by MirceaKitsune** [comment]
@Izzz31 Not always: I tested again and apparently, account X may get the issue when following account Y but not the other way around. What's even weirder is that one of the affected instance interactions seems to have solved itself, but I have no idea what triggered it.

**Comment by ThibG** [comment]
Follows require two-way communication (you sending the follow request, and the other instance accepting it), so there's more room for failure than other kind of interactions.

Furthermore, if your account is silenced on the recipient's instance, sending a follow to even a non-locked user will require approval from them (this is new in 3.0).

**Comment by MirceaKitsune** [comment]
In this case my account isn't silenced, I tried testing the problem with @c-chell yesterday on their instance from it. The only thing that could perhaps be relevant is that the instance from which I'm noticing the issue is masked: This should only mean that its posts are hidden from the public timeline of their instance, but otherwise it should not affect following as far as I'm aware.

Further more, when I attempt to follow a bot I own on the instance, I'm not seeing any follow requests on the account of that bot nor any notification nor did I receive any email on the address the bot is registered with: I've been waiting for +24 hours and checked the spam folder too, and I do receive notification emails whenever anyone mentions my bot so it's not a mail delivery problem with it. There's essentially nothing I can do to make two affected accounts follow each other even when I own and have full access to both of them... it seems the followed account is just not receiving any signal that the follower account is trying to follow it.

**Comment by MirceaKitsune** [comment]
Update: The admin of equestria.social confirmed there are no limitations placed on mastodon.host from which I also reproduced the issue yesterday. baraag.net is masked and I'm getting it there too, which is why for a moment I suspected that could be related after all. If a fully unrestricted instance where my account isn't silenced also produces it, there must be something else.

Even if this was related to silenced accounts though, an issue clearly exists somewhere: Even if a normal follow is turned into a follow request, the followed account would still be receiving that request... in testing with my bot I can confirm the bot sees nothing when I follow it.

**Comment by trwnh** [comment]
>the instance from which I'm noticing the issue is masked: This should only mean that its posts are hidden from the public timeline of their instance, but otherwise it should not affect following as far as I'm aware.

this is a silence. your instance is silenced, which means that all users from that instance are silenced by default. mastodon has two levels of moderation:

- silence, which removes from notifications and public timelines, but you can still follow
- suspend, which deletes/rejects all information

there is also an option to reject only media.

i would say the root cause of your issue is that the admin in question was forced to silence a domain, which is an overly broad action. this could have been avoided if there was an option to *only* remove posts from public timelines.

**Comment by MirceaKitsune** [comment]
@trwnh It doesn't seem to be either instance masking or account silencing though. We tested and this also happens between instances and accounts that place no restrictions on one another.

To be even more sure of this, I again contacted the admins of two instances between which I'm also getting the issue. I asked both of them if the other instance has any restrictions placed on their instance, also if both my or the followed account do such as being silenced. They both confirmed this is not the case and no instance or account restrictions are involved in either direction.

Note that for this latest test, one instance is running vanilla Mastodon v3.0.1 while the other runs Mastodon v2.8.4+glitch (IIRC Mastodon Glitch is a fork). This shouldn't make a difference though, it's the exact same thing I saw with yet another two unrestricted instances yesterday both v3.0.1.

**Comment by trwnh** [comment]
>Note that for this latest test, one instance is running vanilla Mastodon v3.0.1 while the other runs Mastodon v2.8.4 [...] This shouldn't make a difference though

It does make a difference, given that the "convert silenced follows to follow request" behavior was merged in 3.0 https://github.com/tootsuite/mastodon/pull/11975 -- which instance was running 2.8.4? I would assume the one doing the following, no?

**Comment by MirceaKitsune** [comment]
@trwnh It seems to happen both ways in this case: If I follow an account on instance X from instance Y it won't work, but it also won't work if I follow an account on instance Y from instance X. Again I spoke with both admins, the two instances place no restrictions on one another nor my accounts. There seem to be at least two aspects involved:

- Follows are converted to follow requests even when neither the instance or account is silenced.
- Even when this conversion happens, the account being followed doesn't receive that request at all: It simply sees nothing, even if you're told there's a request pending.

**Comment by ThibG** [comment]
Let's clear up a few things:
- Protocol-wise, there is no direct-follow, only follow requests.
- Until 3.0, non-locked accounts would automatically accept all follow requests (unless the requester was blocked), but between two instances, that is still a follow request that has to go from A to B, then an Accept that goes from B to A
- From 3.0 on, a request from a silenced requester is never auto-accepted (but it still gets added to pending follows)

Basically, when you, on instance A, follow someone on instance B, even if their account is not locked and noone is blocked/silenced, the following should happen:
1. A sends a `Follow` request to B
2. B processes the follow request and sends an `Accept` to A
3. A processes the Accept and reflects it internally

From what you're describing, one of these steps isn't happening. It may be that A has issues reaching B, or the other way around. Not enough info was given here to conclude.

**Comment by MirceaKitsune** [comment]
@ThibG Thanks for the info, and that sounds about right. Perhaps the change in 3.0 to treat requests from silenced accounts differently is malfunctioning, affecting both non-silenced accounts but also not properly sending the request through. I'm afraid I can't test any further as I don't own an instance myself and don't wish to annoy other admins with this problem constantly. Maybe someone else is seeing this and can offer more details about their experience?

**Comment by MirceaKitsune** [comment]
Okay, there's clearly a major issue here: I just discovered that this even happens with some accounts on my local instance! If I try to follow a particular account on the same instance as me, it once more becomes a pending request that never succeeds.

The account is not locked, I was able to follow it instantly from another instance. It's also not blocking me or anything, I'm able to interact with their posts just fine and they don't even know me anyway. It's not an issue with my account either, as once more I'm able to follow other users both locally and remotely from it, in this case it's just one user I'm getting the issue with even locally.

**Comment by trwnh** [comment]
Are you silenced on your local instance?

**Comment by MirceaKitsune** [comment]
@trwnh Not that I'm aware of... I'll ask the admin just to be extra sure. My posts appear in the instance's local timeline though, so that should be further confirmation.

In any case, I now seem to be getting this whenever I try following any account on the local instance. This didn't happen a few days ago.

**Comment by MirceaKitsune** [comment]
Update: The admin also took the intuitive decision to restart the server and let me know about it. Once he did that the problem solved itself, I can now follow local accounts again. I wonder if on other instances this also breaks after a certain amount of uptime, or if it's permanent in their case... hoping this piece of info helps though.

**Comment by MirceaKitsune** [comment]
I asked the admin of equestria.social to do a server restart as well, after which I attempted to follow my bot there from baraag.net again. This time restarting did nothing: My attempt to follow the bot is still converted to a request, while the bot receives no email or notification.

EQS does mask Baraag, so in this case the change in 3.0.1 is partly involved. The issue is that the bot still doesn't receive even that follow request.

In yesterday's case however, when the problem began occurring locally, restarting the server fixed it.

**Comment by MirceaKitsune** [comment]
It appears that as of Mastodon 3.1.0 this was fixed. The system to convert follows into follow requests on silenced instances is still there, however the followed account actually sees the notifications and is able to accept them now! I tested this with my bot and can confirm the notifications are seen and possible to approve this time. Closing this and will open another issue if I notice any other problems with following.
